modules/library/render/src/main/java/org/geotools/renderer/label/LabelCacheImpl.java

```diff
@@ -1259,59 +1260,18 @@ private boolean paintPolygonLabel(LabelPainter painter, AffineTransform tempTran
             return false;
         }
         
-        Point centroid;
-        try {
-            centroid = geom.getCentroid();
-        } catch (Exception e) {
-            // generalized polygons causes problems - this
-            // tries to hide them.
-            try {
-                centroid = geom.getExteriorRing().getCentroid();
-            } catch (Exception ee) {
-                try {
-                    centroid = geom.getFactory().createPoint(geom.getCoordinate());
-                } catch (Exception eee) {
-                    return false; // we're hooped
-                }
-            }
-        }
+        Point centroid = RendererUtilities.getPolygonCentroid(geom);
```

Comment:
The code used to return false in case of failure, now centroid gets set to null and then will NPE on the prepared geometry check?